### PR TITLE
[Model] Openvla support

### PR DESCRIPTION
**Pull request by yiwen101**
## Purpose
This PR works on subgoal of issue #42100, adding support for OpenVLA. It is a continuation of PR #32390 by @PalmDr. 

### Transformers backend incompatibility
- OpenVLA's Hugging Face remote code targets older Transformers/tokenizers behavior (`transformers==4.40.1`,`tokenizers==0.19.1`, `timm==0.9.10`). Under current vLLM dependencies, the remote code cannot run without significant compatibility patch (e.g. different method signature, missing fields, missing `GenerationMixin` inheritance ... ). 
- After force-load under the current vLLM dependency stack with monkey patches, representative loaded weights differed from the known-good OpenVLA reference environment.
- The vision side is fused DINOv2 + SigLIP. These towers are created with timm model ids. 
- OpenVLA preprocessing is model-specific: one RGB observation is converted into a 6-channel tensor, with 3 channels normalized for DINOv2/ImageNet and 3 channels normalized for SigLIP.

This diagram by @PalmDr in the PR #32390 illustrates the OpenVLA-7B Architecture
```
┌─────────────┐   ┌─────────────┐
│   DINOv2    │   │   SigLIP    │   Vision Encoders
│  (1024-dim) │   │  (1152-dim) │
└──────┬──────┘   └──────┬──────┘
       └────────┬────────┘
                ▼
        ┌───────────────┐
        │ Concat (2176) │   Fused Features
        └───────┬───────┘
                ▼
        ┌───────────────┐
        │  MLP Projector│   3-layer, GELU
        │  → 4096-dim   │
        └───────┬───────┘
                ▼
        ┌───────────────┐
        │  Llama-2-7B   │   Language Model
        └───────┬───────┘
                ▼
        ┌───────────────┐
        │ 7 Action Tkns │   Robot Control
        └───────────────┘
```
- OpenVLA's processor does not provide the multimodal token-count

### Timm Dependency
This PR currently treats timm as a lazy, model-specific runtime dependency. Manual verification used `timm==0.9.10`, matching the OpenVLA reference environment.

### Scope
While the openVla configs contains fields like time ids, image size, time activation override, they are not actually configurable.

This PR only support and tested OpenVLA-7B variant:
- timm ids:
  - `vit_large_patch14_reg4_dinov2.lvd142m`
  - `vit_so400m_patch14_siglip_224`
- image size: `[224, 224]`
- timm activation overrides: `[None, None]`
- use fused DINOv2 + SigLIP backbone

This Pr contains the above fields in config.py to mirror the HF config, but added guard clause in model implementation to ensure their value are fixed.

### Prompt handling
OpenVLA uses the plain text prompt format `In: ...\nOut:`. The image placeholder tokens are inserted by the multimodal processor after BOS.  I initially tested a literal `<image>` prompt path, but it damaged the HF-vs-vLLM generated-token parity. The current implementation uses `PromptInsertion` after BOS to match the OpenVLA reference prompt shape.

## Test Plan
Parity check in seperate environment was run using 10 sampled [LIBERO](https://github.com/Lifelong-Robot-Learning/LIBERO) observations for both offline and vllm server.`timm==0.9.10` is used in both environments for the OpenVLA vision towers. Unit test on processor is also added.

As the generated token sequences do not always agree, the offline run script dived deeper to compared stage by stage trace and injected saved HF features result into the vLLM.

The temporary test script was removed from the final PR is last commit to keep the upstream change focused.

## Test Result
Local CPU:

```text
tests/models/multimodal/processing/test_openvla.py + tests/models/test_registry.py
8 passed, 367 deselected

tests/models/multimodal/processing/test_common.py -k openvla
3 skipped, 417 deselected

ruff-check
Passed

ruff-format
Passed
```

Summary of manual verification result:

- 6 / 10 cases matched exactly at the generated-token level.
- The first non-zero numerical difference appeared inside the timm DINOv2/SigLIP vision tower execution, then accumulated through later vision blocks.
- The vLLM projector matched HF exactly when fed the same HF fused vision features, so the projector weight loading and projector computation matched the reference.
- Loaded weights matched between HF and vLLM after normalizing known timm key naming differences.
- `pixel_values` matched exactly, so the vLLM processor is producing the same image tensor as the reference path.
- served-vs-HF produced the same exact generated-token parity: 6 / 10.

Numerical magnitude of difference is: 
- dino_features: `mean_abs_diff=0.050265`, `p99_abs_diff=0.1875`, `max_abs_diff=2.5`, 
- siglip_features: `mean_abs_diff=0.020650`, `p99_abs_diff=0.0859375`, `max_abs_diff=1.625`, 

The evidence points to dependency/runtime numerical drift in the timm vision backbone path. I could upload the full run log in comment if they are helpful.

## AI assistance
Codex was used while developing and testing this PR. I reviewed the generated changes and manual verification results.

### CODE REVIEW

File: vllm/model_executor/models/openvla.py [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+        if self.vision_backbone.dinov2_featurizer is None:
+            raise RuntimeError("OpenVLA vision backbone is not initialized.")
+
+        pixel_values = image_input["data"].to(
+            dtype=self.vision_backbone.dinov2_featurizer.patch_embed.proj.weight.dtype
+        )
+        vision_features = self.vision_backbone(pixel_values)
+        return self.projector(vision_features)
+
+    def embed_multimodal(self, **kwargs: object) -> MultiModalEmbeddings:
```
Comment (yiwen101) [inline vllm/model_executor/models/openvla.py:464]: This is a maybe trivial question. I see all other model follow this pattern when implementing embed_multimodal method. But if I inline the two sub methods, add remove "class OpenVLAImagePixelInputs(TensorSchema)", it seems the code will be cleaner.

Comment (DarkLight1337) [reply vllm/model_executor/models/openvla.py:464]: Unfortunately we rely on `_parse_and_validate_image_input` to execute input validation tests automatically. Hope we could improve this in the future...

File: tests/models/registry.py [hunk 2]
```diff
@@ -1198,6 +1194,18 @@ def check_available_online(
             )
         },
     ),
+    "OpenVLAForActionPrediction": _HfExamplesInfo(
+        "openvla/openvla-7b",
+        trust_remote_code=True,
+        max_transformers_version="4.40",
```
Comment (DarkLight1337) [inline tests/models/registry.py]: Is this just for tests? Since I doubt you could run vLLM with such an old Transformers version. If so, I think you could remove this since you already skip the test explicitly in the previous files

File: vllm/model_executor/models/openvla.py [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+
+        raw = np.asarray(image, dtype=np.float32) / 255.0
+        dinov2_pixels = ((raw - self.IMAGENET_MEAN) / self.IMAGENET_STD).transpose(
+            2, 0, 1
+        )
+        siglip_pixels = ((raw - self.SIGLIP_MEAN) / self.SIGLIP_STD).transpose(2, 0, 1)
+        pixel_values = np.concatenate([dinov2_pixels, siglip_pixels], axis=0)
+        return torch.from_numpy(pixel_values)
+
+    def _call_hf_processor(
```
Comment (DarkLight1337) [inline vllm/model_executor/models/openvla.py]: Could you move the HF processing code to `transformers_utils.processors`?

File: vllm/model_executor/models/openvla.py (hunk 1 above)
Comment (yiwen101) [reply vllm/model_executor/models/openvla.py:464]: Got it, thank you for your detailed response!

File: tests/models/registry.py (hunk 2 above)
Comment (yiwen101) [reply tests/models/registry.py]: Got it, sorry for the confusing line. There will be no HF-runner test that execute OpenVLA’s old remote code. I have removed this in latest commit.

File: vllm/model_executor/models/openvla.py (hunk 3 above)
Comment (yiwen101) [reply vllm/model_executor/models/openvla.py]: Thank you for the catch. Refactor done and retested in latest commit.

File: vllm/transformers_utils/processors/openvla.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    if image.dtype != np.uint8:
+        image = image.astype(np.float32)
+        if image.max(initial=0.0) <= 1.0:
+            image = image * 255.0
+        image = np.clip(image, 0, 255).astype(np.uint8)
+
+    return Image.fromarray(image).convert("RGB")
+
+
+def preprocess_openvla_image(image: Any, image_size: int) -> torch.Tensor:
```
Comment (DarkLight1337) [inline vllm/transformers_utils/processors/openvla.py:54]: Maybe refactor this to use torchvision-based preprocessing (see QwenVLProcessor for example)

File: vllm/model_executor/models/openvla.py [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+
+@MULTIMODAL_REGISTRY.register_processor(
+    OpenVLAMultiModalProcessor,
+    info=OpenVLAProcessingInfo,
+    dummy_inputs=OpenVLADummyInputsBuilder,
+)
+class OpenVLAForActionPrediction(nn.Module, SupportsMultiModal, SupportsPP):
+    """OpenVLA wrapper with vLLM language-model execution wired in."""
+
+    embed_input_ids = SupportsMultiModal.embed_input_ids
```
Comment (DarkLight1337) [inline vllm/model_executor/models/openvla.py]: No need to set it explicitly since it  should be inherited

File: vllm/model_executor/models/openvla.py [hunk 6]
```diff
@@ -0,0 +1,551 @@
...
+            if self.use_fused_vision_backbone:
+                self.siglip_featurizer = timm.create_model(
+                    self.timm_model_ids[1],
+                    pretrained=False,
+                    num_classes=0,
+                    img_size=self.image_size,
+                    act_layer=self.timm_override_act_layers[1],
+                )
+
+        self.dinov2_featurizer = self.dinov2_featurizer.to(
```
Comment (DarkLight1337) [inline vllm/model_executor/models/openvla.py]: No need to change the device, it should be set automatically by vLLM

File: vllm/model_executor/models/openvla.py (hunk 5 above)
Comment (yiwen101) [reply vllm/model_executor/models/openvla.py]: Thank you for the catch, removed in latest commit.

File: vllm/model_executor/models/openvla.py (hunk 6 above)
Comment (yiwen101) [reply vllm/model_executor/models/openvla.py]: Got it. I had wrongly put the timm model_create in load_weight. It has been shifted to __init__ in the latest commit.

File: vllm/transformers_utils/processors/openvla.py (hunk 4 above)
Comment (yiwen101) [reply vllm/transformers_utils/processors/openvla.py:54]: Thank you so much for the catch, I have refactored it following QwenVLProcessor example in latest commit.

**Review by DarkLight1337 — APPROVED** [review]
Thanks, let's see if the test passes

**Comment by yiwen101** [comment]
> Thanks, let's see if the test passes

Thank you. I fixed openVLA related CI issue and rebased this PR in last commit because Mergify could not auto-update the branch after upstream changed `.github/workflows/scripts/build.sh`.

The remaining `language-models-tests-hybrid-1` failure appears unrelated to this PR. It fails in the text-only Granite hybrid test, and recent `main` runs show the same job/test failing as well, e.g. buildkite build [66875](https://buildkite.com/vllm/ci/builds/66875/list). Please let me know if you would like me to take any action here

**Comment by DarkLight1337** [comment]
I'll just force merge it
